### ISSUE TITLE
[Suggestion] Items should have a tooltip for their set bonus

### ISSUE THREAD

**Issue by vizthex123** [Laike-Endaril/Set-Bonus#15]
Title. It's kinda annoying to not see what they do, especially if it just adds attributes.

Maybe if there's too much text it only shows up when holding shift?

Or prints a message in the chat whenever you equip a full set?

**Comment by democat3457** [comment]
You can specify text that appears underneath the bonus when the set bonus is completed that usually states what the set does.

**Comment by vizthex123** [comment]
Where's that in the config? The only rendering option I've found seems to do nothing.

**Comment by democat3457** [comment]
There is this bit up here in the client settings that has to be enabled:
```
"client settings" {
    # Whether or not to show set bonus information in discovered set item tooltips
    B:"1. Enable Tooltips"=true
}
```
and then when declaring your bonuses in section 2f, you can do something like this:
```
IceScaleArmorBonus, [+5 Hearts] [Resistance] [+5 Armor], 0, IceScaleArmor
```

**Comment by vizthex123** [comment]
So if I format the diamond set bonus I have like this:

`DArmourProc, Diamond Armour, 1, [+30% Attack Speed] [+2 Armour Toughness], DArmour`

it'll render that in the tooltip?

**Comment by democat3457** [comment]
No, it'd be more like
```
DArmourProc, [+30% Attack Speed] [+2 Armour Toughness], 1, DArmour
```
The format is
`ArmorBonus, SetBonus Tooltip, Discovery Mode, ArmorSet`

**Comment by vizthex123** [comment]
ah ok. Will try it out, thanks.

**Comment by Laike-Endaril** [comment]
Technically that 2nd argument was intended as a name, but in the end it's really just tooltip text, so you can certainly use it that way.

There used to be more detailed tooltips, but I removed them because...I honestly don't remember the reason(s).  It was a long time ago.  Maybe they were just too wordy, or maybe there was another issue.  I know the same item can qualify for multiple sets, so wordiness was certainly part of it.  I also had started a "bonus screen" or "bonus menu" where you could view more detailed information on your known sets / bonuses, but that got put to the side in favor of working on other mods.

In any case, one of my indev mods *might* end up including Set Bonus's featureset, so I've stopped doing feature updates on Set Bonus until I know whether it's getting absorbed by the indev mod or not.

I'll leave this open for issue visibility and future reference.

**Comment by vizthex123** [comment]
Alright, thanks. I do hope someone is able to update the mod and make it more clear how to add tooltips for what a set bonus gives (or maybe it could be automatic)

**Comment by vizthex123** [comment]
Alright I tested it, doing`DArmourProc, Diamond Armour, [+30% Attack Speed] [+2 Armour Toughness], 1, DArmour` does nothing. No tooltips render other than the regular set bonus name one.

**Comment by Laike-Endaril** [comment]
`DArmourProc, Diamond Armour, [+30% Attack Speed] [+2 Armour Toughness], 1, DArmour`
Should throw an error, but...

`DArmourProc, Diamond Armour [+30% Attack Speed] [+2 Armour Toughness], 1, DArmour`
Should work, in theory.  Just keep in mind that it won't format it nicely.  The mod doesn't *officially* support adding to the tooltip this way, so it will not format it very nicely.

Now...there's a little hack you can use to add newlines into it (only because FLib has a special tag that it detects for tooltips in general)...
`DArmourProc, FIXNEWLINES Diamond Armour\n[+30% Attack Speed]\n[+2 Armour Toughness], 1, DArmour`

The `FIXNEWLINES` bit gets removed automatically, and converts any instances of `\n` to actual newlines.  These lines won't retain the text color due to the way MC handles text color, so they'll be white, but it's still probably better than the alternative.

**Comment by vizthex123** [comment]
Ah ok, thanks. Will try it out.

Hope the mod can get updated soon. I have to check the config for set bonuses and it's kind of annoying.

**Comment by democat3457** [comment]
can be closed probably

**Comment by Laike-Endaril** [comment]
The newest versions have much better tooltip support.  I think I'll finally actually close this :P
